packages/components/src/components/badge/badge.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import classNames from 'classnames';
+
+type TBadgeSize = 'medium' | 'large';
+type TBadgeSpacing = 'tight' | 'loose';
+type TWeight = 'bold' | 'regular';
+type TBackgroundColor = 'blue' | 'orange' | 'red' | 'gray';
+type TRoundedCorners = 4 | 2;
+
+interface Badge extends React.HTMLAttributes<HTMLSpanElement> {
+    className?: string;
+    color?: string;
+    label: string;
+    rounded_corners?: TRoundedCorners;
+    size?: TBadgeSize;
+    spacing?: TBadgeSpacing;
+    weight?: TWeight;
+}
+
+interface BadgeContained extends Badge {
+    type: 'contained';
+    background_color: TBackgroundColor;
+}
+
+interface BadgeBordered extends Badge {
+    type: 'bordered';
+}
+
+type BadgeProps = BadgeContained | BadgeBordered;
+
+const Badge = (props: BadgeProps) => {
+    const {
+        className,
+        color,
+        label,
+        rounded_corners = 4,
+        size = 'medium',
+        spacing = 'tight',
+        type,
+        weight = 'bold',
+        ...rest
```

Comment:
```suggestion
        background_color,
        ...rest
```

Reply:
@mahdiyeh-deriv since i use Discriminated Unions for badge type ('bordered' | 'contained') in this component, i should hide background_color into rest props, otherwise typescript will complain that background_color doesn't exist in 'bordered' type. This is a part of implementation of Discriminated Unions pattern. If we decide to not use this pattern then i will refactor `background_color` here. But i need confirmation first. Pls check [this](https://github.com/binary-com/deriv-app/pull/8384#discussion_r1182130721) first